app/WebParsing/PostParser.hs

```diff
@@ -63,37 +65,32 @@ addPostCategoriesToDatabase postCode categories = do
                 ((length string) >= 7) && ((length $ filter (\bool -> bool) infixes) <= 0)
         containsString string substring = isInfixOf substring string
 
-addCategoryToDatabase :: String -> String -> IO ()
+addCategoryToDatabase :: T.Text -> String -> SqlPersistM ()
 addCategoryToDatabase postCode category =
-    insertPostCategory (T.pack category) (T.pack postCode)
+    insert_ $ PostCategory (T.pack category) postCode
 
 
 -- Helpers
 
-generalParser :: [Tag String] -> Maybe String -> T.Text -> IO ()
-generalParser tags firstCourse postCode = do
-    let parsed = P.parse (generalCategoryParser firstCourse) "Failed." (innerText tags)
+categoryParser :: [Tag String] -> Maybe String -> T.Text -> [[Tag String]] -> SqlPersistM ()
+categoryParser tags firstCourse postCode liPartitions = do
     case parsed of
         Right (description, departmentName, postType, categories) -> do
-            insertPost (T.pack departmentName) (T.pack postType) postCode (T.pack description)
-            addPostCategoriesToDatabase (T.unpack postCode) categories
+            addPostCategoriesToDatabase postCode categories
+            insert_ $ Post postType departmentName postCode description
         Left message -> do
-            print message
-
-liParser :: [Tag String] -> [[Tag String]] -> Maybe String -> T.Text -> IO ()
-liParser tags liPartitions firstCourse postCode = do
-    let categories = map parseLi liPartitions
-        postInfo = P.parse (postInfoParser firstCourse) "Failed." (innerText tags)
-    case postInfo of
-        Right (description, departmentName, postType) -> do
-            insertPost (T.pack departmentName) (T.pack postType) postCode (T.pack description)
-            addPostCategoriesToDatabase (T.unpack postCode) categories
-        Left message -> do
-            print message
+            return ()
+    where
+        parsed = case liPartitions of 
+            [] -> P.parse (generalCategoryParser firstCourse) "Failed." (innerText tags)
+            partitions -> do
+                let categories = map parseLi partitions
+                (description, departmentName, postType) <- P.parse (postInfoParser firstCourse) "Failed." (innerText tags)
```

Comment:
@christinem Just a reminder that the last thing to fix for this PR is the return type of the parsers. After that I'd like to merge this in.